pkg/pubsub/node_event_tracker.go

```diff
@@ -135,30 +135,53 @@ func (net *NodeEventTracker) ListenClose(n network.Network, a ma.Multiaddr) {
 // saves the updated nodeData.
 func (net *NodeEventTracker) Connected(n network.Network, c network.Conn) {
 	// A node has joined the network
+	logrus.Debug("Connected")
+	if net == nil {
+		logrus.Error("[-] Network is nil")
+		return
+	}
+
+	// check if the connection is nil
+	if c == nil {
+		logrus.Error("[-] Connection is nil")
+		return
+	}
 	remotePeer := c.RemotePeer()
+
+	// get the peer ID of the remote peer
 	peerID := remotePeer.String()
 
 	nodeData, exists := net.nodeData.Get(peerID)
 	if !exists {
 		// WTF: Shouldn't we add it? We don't yet have the NodeData but we can at least add it.
 		return
 	} else {
-		if nodeData.IsActive {
-			// Node appears already connected, buffer this connect event
-			net.ConnectBuffer[peerID] = ConnectBufferEntry{NodeData: nodeData, ConnectTime: time.Now()}
-		} else {
-			nodeData.Joined(net.nodeVersion)
-			err := net.AddOrUpdateNodeData(nodeData, true)
-			if err != nil {
-				logrus.Error("[-] Error adding or updating node data: ", err)
-				return
+		if nodeData != nil {
```

Comment:
In principle this should never happen thanks to the `if !exists` above (except if at some point we actually added a `nil` because of a bug).